### PR TITLE
Harbor

### PR DESCRIPTION
**Pull request by jcotoBan**
Updating harbor.yaml file that has issues and converting harbor to ansible script.

### CODE REVIEW

File: apps/linode-marketplace-harbor/roles/harbor/templates/motd.j2 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+*********************************************************
+Akamai Connected Cloud Harbor Marketplace App
+App URL: https://{{ _domain }}
+Credentials File: /root/.credentials
```
Comment (hmorris3293) [inline apps/linode-marketplace-harbor/roles/harbor/templates/motd.j2:4]: It doesn't appear that this creds file is generated in the playbook. You can copy what we did for the other apps
https://github.com/akamai-compute-marketplace/marketplace-apps/blob/main/apps/linode-marketplace-joplin/roles/joplin/tasks/main.yml#L66

File: deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+#<UDF name="disable_root" label="Disable root access over SSH?" oneOf="Yes,No" default="No">
+#<UDF name="pubkey" label="The SSH Public Key that will be used to access the Linode (Recommended)" default="">
+
+## Domain Settings
+#<UDF name="token_password" label="Your Linode API token. This is needed to create your WordPress server's DNS records" default="">
+#<UDF name="subdomain" label="Subdomain" example="The subdomain for the DNS record: www (Requires Domain)" default="">
+#<UDF name="domain" label="Domain" example="The domain for the DNS record: example.com (Requires API token)" default="">
+
+## harbor Settings 
+#<UDF name="harbor_admin_password" label="The Harbor admin password">
```
Comment (hmorris3293) [inline deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh:17]: Can we generate a password for this in the playbook and put it in /root/credentials? You can copy what we do for the mysql helper

https://github.com/akamai-compute-marketplace/marketplace-apps/blob/main/apps/linode_helpers/roles/securemysql/tasks/passgen.yml#L6

File: deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+#<UDF name="harbor_db_password" label="The Harbor database password ">
```
Comment (hmorris3293) [inline deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh:18]: same as above comment

File: deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+}
+
+function run {
+  # install dependancies
+  apt-get update
+  apt-get install -y git python3 python3-pip
+
+  # clone repo and set up ansible environment
+  git -C /tmp clone --branch harbor ${GIT_REPO}
```
Comment (hmorris3293) [inline deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh:98]: Need to remove `--branch harbor`

File: deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+#<UDF name="subdomain" label="Subdomain" example="The subdomain for the DNS record: www (Requires Domain)" default="">
+#<UDF name="domain" label="Domain" example="The domain for the DNS record: example.com (Requires API token)" default="">
+
+## harbor Settings 
+#<UDF name="harbor_admin_password" label="The Harbor admin password">
+#<UDF name="harbor_db_password" label="The Harbor database password ">
+#<UDF name="soa_email_address" label="Admin Email for the Harbor server and Let's Encrypt SSL certificate">
+
+# git repo
+export GIT_REPO="https://github.com/jcotoBan/marketplace-apps.git"
```
Comment (hmorris3293) [inline deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh:22]: needs to be updated to https://github.com/akamai-compute-marketplace/marketplace-apps.git"

File: deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh (hunk 4 above)
Comment (jcotoBan) [reply deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh]: Done

File: deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh (hunk 5 above)
Comment (jcotoBan) [reply deployment_scripts/linode-marketplace-harbor/harbor-deploy.sh]: Done

**Review by hmorris3293 — APPROVED** [review]

**Review by hmorris3293 — APPROVED** [review]

**Pull request merged**